common/beerocks/bcl/include/bcl/beerocks_promise.h

```diff
@@ -22,53 +24,57 @@ template <class T> class promise {
     // Set value for waiting threads
     void set_value(const T value)
     {
-        pthread_mutex_lock(&m_mut);
+        int err = pthread_mutex_lock(&m_mut);
+        mapf_assert(err == 0);
         m_value  = value;
         m_signal = 1;
-        pthread_cond_broadcast(&m_cond);
-        pthread_mutex_unlock(&m_mut);
+        err      = pthread_cond_broadcast(&m_cond);
+        mapf_assert(err == 0);
+        err = pthread_mutex_unlock(&m_mut);
+        mapf_assert(err == 0);
     }
-
     // Wait for a value to be set
     void wait()
     {
         // lock the mutex and wait for the conditional variable to be set
-        pthread_mutex_lock(&m_mut);
+        int err = pthread_mutex_lock(&m_mut);
+        mapf_assert(err == 0);
         // If signaled before the wait, no need to wait
         if (m_signal == 1) {
             m_signal = 0;
         } else {
-            pthread_cond_wait(&m_cond, &m_mut);
+            err = pthread_cond_wait(&m_cond, &m_mut);
+            mapf_assert(err == 0);
         }
-        pthread_mutex_unlock(&m_mut);
+        err = pthread_mutex_unlock(&m_mut);
+        mapf_assert(err == 0);
     }
 
     // Wait for a value to be set with timeout
     bool wait_for(uint32_t timeout_ms)
     {
         struct timeval now;
         struct timespec timeout;
-
         // set the absolut timeout
         gettimeofday(&now, NULL);
         timeout.tv_sec  = now.tv_sec + (timeout_ms / 1000);
         timeout.tv_nsec = (now.tv_usec + (timeout_ms % 1000) * 1000) * 1000;
 
-        int retcode = 0;
-
         // lock the mutex and wait for the conditional variable to be set or timeout
-        pthread_mutex_lock(&m_mut);
+        int err = pthread_mutex_lock(&m_mut);
+        mapf_assert(err == 0);
         // If signaled before the wait, no need to wait
         if (m_signal == 1) {
             m_signal = 0;
         } else {
-            retcode = pthread_cond_timedwait(&m_cond, &m_mut, &timeout);
+            if ((err = pthread_cond_timedwait(&m_cond, &m_mut, &timeout)) != 0) {
+                LOG(ERROR) << "pthread_cond_timedwait failed, error code: " << err;
```

Comment:
(nitpick) it may be useful to identify the failure at a glance, so adding `strerror(err)` to the log is definitely useful, in my experience (tracking down the correct version of errno.h and cross-referencing error numbers can be time consuming)